cpp/src/strings/search/find.cu

```diff
@@ -414,6 +422,304 @@ std::unique_ptr<column> contains_warp_parallel(strings_column_view const& input,
   return results;
 }
 
+/**
+ * Each string uses a warp(32 threads) to handle all the targets.
+ * Each thread uses num_targets bools shared memory to store temp result for each lane.
+ */
+CUDF_KERNEL void multi_contains_warp_parallel_multi_scalars_fn(
+  column_device_view const d_strings,
+  column_device_view const d_targets,
+  cudf::device_span<char> const d_target_first_bytes,
+  column_device_view const d_target_indexes_for_first_bytes,
+  cudf::device_span<bool*> d_results)
+{
+  auto const num_targets = d_targets.size();
+  auto const num_rows    = d_strings.size();
+
+  auto const idx = static_cast<size_type>(threadIdx.x + blockIdx.x * blockDim.x);
+  if (idx >= (num_rows * cudf::detail::warp_size)) { return; }
+
+  auto const lane_idx = idx % cudf::detail::warp_size;
+  auto const str_idx  = idx / cudf::detail::warp_size;
```

Comment:
The `num_rows * cudf::detail::warp_size` can overflow `size_type`
```suggestion
  auto const idx         = cudf::detail::grid_1d::global_thread_id();
  auto const str_idx     = idx / cudf::detail::warp_size;
  if (str_idx >= num_rows) { return; }

  auto const lane_idx = idx % cudf::detail::warp_size;
```

Reply:
Done.